Sources/Fullscreen/TransactionView/TransactionView.swift

```diff
@@ -0,0 +1,181 @@
+        return stackView
+    }()
+
+    // MARK: - Init
+
+    public init(title: String, dataSource: TransactionViewDataSource, delegate: TransactionViewDelegate, withAutoLayout autoLayout: Bool = true) {
+        self.dataSource = dataSource
+        self.delegate = delegate
+
+        super.init(frame: .zero)
+
+        self.numberOfSteps = dataSource.transactionViewNumberOfSteps(self)
+        self.currentStep = dataSource.transactionViewCurrentStep(self)
+
+        titleLabel.text = title
+        translatesAutoresizingMaskIntoConstraints = !autoLayout
+
+        setup()
+        progressTo(currentStep)
+    }
+
+    public required init?(coder aDecoder: NSCoder) { fatalError("init(coder:) has not been implemented") }
+}
+
+// MARK: - Private
+
+private extension TransactionView {
+    func setup() {
+        backgroundColor = .bgPrimary
+
+        addSubview(titleLabel)
+        addSubview(scrollView)
+        scrollView.addSubview(scrollableContentView)
+
+        scrollableContentView.fillInSuperview()
+        scrollableContentView.addSubview(verticalStackView)
+
+        NSLayoutConstraint.activate([
+            titleLabel.leadingAnchor.constraint(equalTo: safeAreaLayoutGuide.leadingAnchor, constant: .largeSpacing),
+            titleLabel.trailingAnchor.constraint(equalTo: safeAreaLayoutGuide.trailingAnchor),
+            titleLabel.topAnchor.constraint(equalTo: safeAreaLayoutGuide.topAnchor, constant: .mediumSpacing),
+            titleLabel.heightAnchor.constraint(equalToConstant: 40),
+
+            scrollView.leadingAnchor.constraint(equalTo: safeAreaLayoutGuide.leadingAnchor),
+            scrollView.trailingAnchor.constraint(equalTo: safeAreaLayoutGuide.trailingAnchor),
+            scrollView.topAnchor.constraint(equalTo: titleLabel.bottomAnchor, constant: .mediumSpacing),
+            scrollView.bottomAnchor.constraint(equalTo: safeAreaLayoutGuide.bottomAnchor),
+
+            scrollableContentView.heightAnchor.constraint(greaterThanOrEqualTo: scrollView.heightAnchor, constant: .mediumSpacing),
+            scrollableContentView.widthAnchor.constraint(equalTo: scrollView.widthAnchor),
+
+            verticalStackView.trailingAnchor.constraint(equalTo: scrollableContentView.trailingAnchor, constant: -.mediumLargeSpacing),
+            verticalStackView.topAnchor.constraint(equalTo: scrollableContentView.topAnchor, constant: .largeSpacing),
+            verticalStackView.bottomAnchor.constraint(equalTo: scrollableContentView.bottomAnchor),
+        ])
+
+        for index in 0..<numberOfSteps {
+            guard let model = dataSource?.transactionViewModelForIndex(self, forStep: index) else { return }
+
+            addTransactionStepView(index, model)
+            addTransactionStepDots(index)
+        }
+
+        guard let stepDot = stepDots.first else { return }
+        verticalStackView.leadingAnchor.constraint(equalTo: stepDot.trailingAnchor, constant: .mediumSpacing).isActive = true
```

Comment:
why not just use `NSLayoutConstraint.activate([])`

Reply:
It's just one anchor, but could use it I guess.

Reply:
most of my comments are minor/ trivial :D
most == all 💯